superset/assets/javascripts/SqlLab/components/SqlEditor.jsx

```diff
@@ -148,7 +148,7 @@ class SqlEditor extends React.PureComponent {
         <div className="pull-left">
           <Form inline>
             <RunQueryActionButton
-              allowAsync={this.props.database && this.props.database.allow_run_async}
+              allowAsync={this.props.database ? this.props.database.allow_run_async : false}
```

Comment:
[not that important] I'm surprised this doesn't cause an `eslint` error, was it complaining about not being a boolean? `!!this.props.database && this.props.database.allow_run_async` or `Boolean(this.props.database) && this.props.database.allow_run_async` could be alternatives.